client/controller.js

```diff
@@ -51,6 +52,10 @@ export function clientRouter( route, ...middlewares ) {
 	page( route, ...[ ...middlewares, render ] );
 }
 
+export function dispatchSetSection( context, section ) {
```

Comment:
An alternative approach would be to make this a middleware generator (with the returned middleware's `next` defaulting to a `noop`/`function() {}`), which could then be directly used [here](https://github.com/Automattic/wp-calypso/pull/4137/files#diff-336276ec8a6d8fa6e6ecd99de5a7bab3R36). But that's somewhat of a personal style preference question I'll leave to you to decide.